arch/risc-v/src/esp32c3/esp32c3_sha.c

```diff
@@ -0,0 +1,1666 @@
+/****************************************************************************
+ * arch/risc-v/src/esp32c3/esp32c3_sha.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#ifdef CONFIG_ESP32C3_SHA_ACCELERATOR
+
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <debug.h>
+#include <semaphore.h>
+
+#include "riscv_arch.h"
+#include "hardware/esp32c3_sha.h"
+#include "hardware/esp32c3_system.h"
+
+#include "esp32c3_sha.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define PUT_UINT32_BE(n,b,i)                      \
+{                                                 \
+  (b)[(i)  ] = (unsigned char) ((n) >> 24);       \
+  (b)[(i) + 1] = (unsigned char) ((n) >> 16);     \
+  (b)[(i) + 2] = (unsigned char) ((n) >>  8);     \
+  (b)[(i) + 3] = (unsigned char) ((n)  );         \
```

Comment:
```suggestion
  (b)[(i) + 3] = (unsigned char) ((n));         \
```

Reply:
Please, fix other similar excessive spaces.

Reply:
Thanks, I have modified it.